### DISCUSSION TITLE
Cannot remove big file from Git Commit to push successfully

### DISCUSSION THREAD

**Discussion by aaaronism** [community/community#70014] [Other Feature Feedback, Questions, & Ideas]
Hi, I've contacted support about this issue but I feel like I'm not getting the help I need. I'm wondering if anyone here knows how to resolve this.
**Things to keep in mind:** 

1. I was not working with Git LFS.
2. I'm working on a React project and the public folder is holding 3d models so those files are pretty big.

**Steps that led to the issue:**

1. I went through git commands to try and make an updated commit to my project using git add --all, then git commit -m "commit", then git push origin master. 
2. It gave me an error saying I had big files (from the public folder) so therefore I cannot push the commit. I then typed the exact filepath and tried the git rm as suggested from the documentation and it said fatal: pathspec 'filepath' did not match any files. 
3. I tried resetting and everything. everytime I go back to git push origin master, it says that same file is there. I even tried hard resetting and changing heads so that I was at the branch before the most recent commit I was trying to push.

**The main issue:**
I cannot get rid of the big files from the public folder to push a successful commit.

**Comment by seyLu** [comment]
#### Hi @aaaronism,

If you're fine with not preserving your changes. 

You can delete the current local repo, reclone, add the large files in `.gitignore`, and add, commit, push as usual.

If you need to preserve changes, a hack would be:
- delete .git directory
- clone repo on another directory
- copy/move .git directory from the newly cloned one to existing one
- add large files to .gitignore
- empty git cache

```bash
git rm -r --cached .
```

- add, commit, push as usual

**Comment by oussamaef06** [comment]
It seems like you're dealing with a situation where large files have been added to your Git repository, and you're encountering issues trying to remove them and push your changes. Here's a step-by-step guide to help you resolve this issue:

1. Identify Large Files:

First, you need to identify the large files that are causing the problem. You can use the git-lfs command to list large files:

git lfs ls-files

2. Install Git LFS (if not already installed):

If you're not working with Git LFS and the large files are already committed, you should consider installing Git LFS if it's not already installed. It will help you manage large files more effectively. Install Git LFS by running:

git lfs install

3. Create a .gitattributes File:

You'll need to create a .gitattributes file in your project's root directory or update it if it already exists. In this file, you can specify which files should be tracked by Git LFS. For example:
vbnet
Copy code
public/*.obj filter=lfs diff=lfs merge=lfs -text
This example tells Git LFS to track .obj files in the public directory.

4. Remove the Large Files:

Once you've configured Git LFS, you can remove the large files from your Git history using the git filter-branch command. This command will rewrite your repository's history to remove the large files:

git filter-branch --index-filter 'git rm --cached --ignore-unmatch public/largefile.obj' HEAD
Replace public/largefile.obj with the path to the large file you want to remove.

5. Push Changes:

After removing the large files, you can force-push the cleaned repository to GitHub. Be very careful when using --force, as it rewrites history. Others working on the repository will need to be aware of this change:

git push origin --force

6. Replace Large Files with Git LFS:

If you want to keep these large files under version control, consider adding them to Git LFS. Use the following commands for each large file you want to add:

git lfs track 'public/largefile.obj'
git add .gitattributes
git add public/largefile.obj
git commit -m "Add large file to Git LFS"

7. Inform Collaborators:

If you are collaborating with others on the repository, inform them about the force-push and any changes you've made to the repository's structure.
Keep in mind that rewriting history with filter-branch and force-pushing should be done with caution, especially if you're working in a team. It can cause synchronization issues for collaborators, so make sure everyone is aware of the changes you're making.
